gen/enum.go

```diff
@@ -96,23 +96,31 @@ func enum(g Generator, spec *compile.EnumSpec) error {
 		<end>
 
 		<$v := newVar "v">
+		<$value := newVar "value">
 		// UnmarshalText tries to decode <$enumName> from a byte slice
 		// containing its name.
 		<- if .Spec.Items>
 		//
 		//   var <$v> <$enumName>
 		//   err := <$v>.UnmarshalText([]byte("<(index .Spec.Items 0).Name>"))
 		<- end>
-		func (<$v> *<$enumName>) UnmarshalText(value []byte) error {
-			switch string(value) {
+		func (<$v> *<$enumName>) UnmarshalText(<$value> []byte) error {
+			<- $s := newVar "s" ->
+			<- $val := newVar "val" ->
+			switch <$s> := string(<$value>); <$s> {
 			<- $enum := .Spec ->
 			<range .Spec.Items ->
 				case "<enumItemLabelName .>":
 					*<$v> = <enumItemName $enumName .>
 					return nil
 			<end ->
 				default:
-					return <$fmt>.Errorf("unknown enum value %q for %q", value, "<$enumName>")
+					<$val>, err := <$strconv>.ParseInt(<$s>, 10, 64)
```

Comment:
given that enums are `int32` (line 79), should this be `ParseInt(<$s>, 10, 32)` (32 bits instead of 64 bits)

Otherwise, we silently truncate 64-bit values to 32-bit